shared/appengine.go

```diff
@@ -46,15 +59,19 @@ func init() {
 		if runtimeIdentity.Version == "" {
 			panic("Missing environment variable: GAE_VERSION")
 		}
-		service, err := apps.NewService(context.Background())
-		if err != nil {
+		if service, err := apps.NewService(ctx); err != nil {
 			panic(err)
+		} else {
+			if runtimeIdentity.application, err = service.Apps.Get(proj).Do(); err != nil {
+				panic(err)
+			}
 		}
-		runtimeIdentity.application, err = service.Apps.Get(proj).Do()
-		if err != nil {
+		runtimeIdentity.LocationID = runtimeIdentity.application.LocationId
+
+		var err error
+		if clients.cloudtasks, err = cloudtasks.NewClient(ctx); err != nil {
```

Comment:
Would adding `defer client.Close()` somewhere make a difference? you mentioned it is bound to the runtime instead of each request. I assume there is one runtime per version of a service? I guess my question is, in the standard environment without a long-running background process, `clients.cloudtasks` won't get initialized multiple times without being `Close()`?

Reply:
That's a great point!

These clients are indeed bound to the runtime / server process. Even in a standard environment, there's still a server process being started and terminated on demand (*not* for each request, but automatically managed by AppEngine); it's just that we cannot rely on this server process to run indefinitely.

We do need to close the clients before exiting though, in order to prevent leaking any resources (connections). I've added that in the latest commit.